pkg/cli/start.go

```diff
@@ -24,58 +32,142 @@ var startCmd = &cobra.Command{
 	the input device specified via the --source flags
 	`,
-	Example: `  telemetry start --schema=/foo.asciipb [--port=port] [--fake] [--tty=/dev/...] [--db=a.db]`,
-	RunE:    runStart,
+	Example: `
+	telemetry start --schema=/foo.asciipb
+									[--port=port]
+									[--source=(s|f|r)]
+									[--rest]
+									[--tty=/dev/...]
+									[--db=a.db]
+	`,
+	RunE: runStart,
 }
 
 // ErrorCode is the value to be used by main() as exit code in case of
 // error. For most errors 1 is appropriate, but a signal termination
 // can change this.
 var ErrorCode = 1
 
-var fake bool
-var serverPort int
-var ttyPort string
-var dbName string
-var schemaFile string
+var envPrefix = "uwmidsuntelem"
 
 func init() {
-	startCmd.Flags().StringVarP(&schemaFile, "schema", "s", "", "s")
-	startCmd.Flags().IntVarP(&serverPort, "port", "p", 8080, "port")
-	startCmd.Flags().BoolVarP(&fake, "fake", "f", false, "fake")
-	startCmd.Flags().StringVarP(&ttyPort, "tty", "t", "/dev/ttyUSB0", "tty")
-	startCmd.Flags().StringVarP(&dbName, "db", "d", "", "db")
+	startCmd.Flags().String("schema", "", "The location of the CAN message schema")
+	startCmd.Flags().Int("serverPort", 8080, "The port to setup the HTTP port on")
+	startCmd.Flags().String("interface", "/dev/ttyUSB0", "The interface to listen on")
+	startCmd.Flags().String("dbConnectString", "", "The connect string to use for the database")
+	startCmd.Flags().String("token", "", "Permanently add a new token to the sqlite auth table")
+	startCmd.Flags().String("source", "s", "Source of CAN messages. Can be a combination of (s)erial, (r)est, or (f)ake")
+	startCmd.Flags().String("dbDriver", "sqlite3", "The DB driver to use")
+	startCmd.Flags().String("remoteurl", "", "If CAN data should be sent to a remote server")
+
+	viper.SetEnvPrefix(envPrefix)
+	viper.AutomaticEnv()
+	viper.BindPFlags(startCmd.Flags())
 }
 
-func setupURLRouting(r *chi.Mux, messageBus *pubsub.MessageBus) {
-	r.Get("/ws", ws.ServeHTTP(messageBus, ttyPort, fake))
+func setupFileServer(r *chi.Mux, endpoint string, filesDir string) {
+	if !strings.HasSuffix(endpoint, "/") {
+			panic("Endpoint must be a folder type")
+	}
+	fileServer := http.StripPrefix(
+		endpoint,
+	  http.TimeoutHandler(http.FileServer(http.Dir(filesDir)), 5 * time.Second, "Failed to serve file"),
+	)
+	log.Infof("Mounting fileserver from %s", filesDir)
+	endpoint += "*"
+	r.Get(endpoint, func (w http.ResponseWriter, r *http.Request) {
+			fileServer.ServeHTTP(w, r)
+		},
+	)
+}
+
+func setupURLRouting(r *chi.Mux, messageBus *pubsub.MessageBus, db *sql.DB) error {
+	ttyPort := viper.GetString("interface")
+	source := viper.GetString("source")
+	serverPort := viper.GetInt("serverPort")
+
+	r.Get("/ws", ws.ServeHTTP(messageBus))
 	workDir, _ := os.Getwd()
 	filesDir := filepath.Join(workDir, "client", "src")
-	r.FileServer("/", http.Dir(filesDir))
+
+	setupFileServer(r, "/", filesDir)
+
+	// Start the rest api if it was requested
+	if strings.Contains(source, "r") {
```

Comment:
This reads a bit unclear but it seems fine for now. If we repeat this kind of logic to check for these symbols elsewhere, it might be better to abstract to some kind of  Source object.